.github/workflows/build_images.yml

```diff
@@ -2,38 +2,63 @@ name: Build Images
 
 concurrency: ${{ github.ref }}
 
-on:
+on: 
   workflow_dispatch:
-  push:
-    branches: [main]
-    paths:
-      - 'images/**'
-      - 'scripts/**'
-      # - 'builder/**'
 
 jobs:
+
   build:
-    # runs-on: windows-latest
     runs-on: ubuntu-latest
 
-    if: "!contains(join(github.event.commits.*.message), '+nobuild')"
-
     env:
-      STORAGE_ACCOUNT: contosoimagesstorage
-      SUBNET_ID: /subscriptions/e5f715ae-6c72-4a5c-87c8-495590c34828/resourceGroups/Contoso-Images/providers/Microsoft.Network/virtualNetworks/contoso-images-vnet/subnets/builders
+      RESOURCE_GROUP_NAME: "aqovia-devbox"
+      IDENTITY_NAME: "aqovia-devcenter-identity"
+      GALLARY_NAME: "aqovia_compute_gallery"
+      VNET_NAME: "vnetBuilder"
+      SUBNET_NAME: "default"
+      STORAGE_ACCOUNT_NAME: "aqdevboxbuilderstrg"
 
     steps:
       - uses: actions/checkout@v2
 
       - name: Login to Azure
         run: az login --service-principal -u ${{ secrets.AZURE_CLIENT_ID }} -p ${{ secrets.AZURE_CLIENT_SECRET }} --tenant ${{ secrets.AZURE_TENANT_ID }}
-
-      # az cli installs bicep on demand the first time it's used
-      # using the --async argument in the command below simultaniously invokes several commands that use bicep
-      # so if bicep isn't already installed, the all initiate the install which throws a file busy error so we
-      # execute an arbitrary az bicep command here (and get the latest updates in the process)
+          
       - name: Ensure Bicep
         run: az bicep upgrade
+      
+      - name: Update predeploy/main.parameters.json
+        uses: restackio/update-json-file-action@v2.0
+        with:
+          file: ./resources/predeploy/main.parameters.json
```

Comment:
I think the folder name is better to be `pre_build` instead of `predeploy`. Same for postdeploy.